dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/ndk/DatadogNdkCrashHandler.kt

```diff
@@ -0,0 +1,231 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.ndk
+
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.domain.Deserializer
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.log.LogAttributes
+import com.datadog.android.log.internal.domain.Log
+import com.datadog.android.log.internal.domain.LogGenerator
+import com.datadog.android.rum.internal.data.file.RumFileWriter
+import com.datadog.android.rum.internal.domain.event.RumEvent
+import com.datadog.android.rum.internal.domain.event.RumEventDeserializer
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonSyntaxException
+import java.io.File
+import java.io.IOException
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.TimeUnit
+
+internal class DatadogNdkCrashHandler(
+    private val ndkCrashDataDirectory: File,
+    private val dataPersistenceExecutorService: ExecutorService,
+    private val asyncLogWriter: Writer<Log>,
+    private val asyncRumWriter: Writer<RumEvent>,
+    private val logGenerator: LogGenerator,
+    private val rumEventDeserializer: Deserializer<RumEvent> = RumEventDeserializer()
+) : NdkCrashHandler {
+
+    override fun handleNdkCrash() {
+        dataPersistenceExecutorService.submit {
+            checkAndHandleNdkCrashReport()
+        }
+    }
+
+    private fun checkAndHandleNdkCrashReport() {
+        if (ndkCrashDataDirectory.exists()) {
+            // check if there is an NDK report
+            val crashLogFiles =
+                ndkCrashDataDirectory.listFiles { _, name -> name == CRASH_LOG_FILE_NAME }
+            if (crashLogFiles != null && crashLogFiles.isNotEmpty()) {
+                readFromFile(
+                    crashLogFiles[0],
```

Comment:
I'd write something along the lines of : 

```kotlin
val crashLogFile = ndkCrashDataDirectory.listFiles { _, name -> 
       name == CRASH_LOG_FILE_NAME 
    }.firstOrNull()

if (crashLogFile != null) {
    val crashLog = readFromFile(crashLogFile, …)
    val lastRumViewEvent = searchLastRumViewEvent()
    handleNdkCrashLog(crashLog, lastRumViewEvent)
}
```

If you let `handleNdkCrashLog()` accept nullable input, it would be much more readable and avoid nesting complex Kotlin sugar.